extensions/amp-list/0.1/amp-list.js

```diff
@@ -494,25 +482,9 @@ export class AmpList extends AMP.BaseElement {
           !this.ssrTemplateHelper_.isEnabled(),
           '[amp-list]: "amp-script" URIs cannot be used in SSR mode.'
         );
-
-        const args = src.slice(AMP_SCRIPT_URI_SCHEME.length).split('.');
-        userAssert(
-          args.length === 2 && args[0].length > 0 && args[1].length > 0,
-          '[amp-list]: "amp-script" URIs must be of the format "scriptId.functionIdentifier".'
-        );
-
-        const ampScriptId = args[0];
-        const fnIdentifier = args[1];
-        const ampScriptEl = this.element
-          .getAmpDoc()
-          .getElementById(ampScriptId);
-        userAssert(
-          ampScriptEl && ampScriptEl.tagName === 'AMP-SCRIPT',
-          `[amp-list]: could not find <amp-script> with script set to ${ampScriptId}`
-        );
-        return ampScriptEl
-          .getImpl()
-          .then((impl) => impl.callFunction(fnIdentifier));
+        return Services.scriptForDocOrNull(
+          this.element
+        ).then((ampScriptService) => ampScriptService.fetch(src));
```

Comment:
`ampScriptService` can be `null`, right?

Reply:
Yeah. I can swap out for a better error message at each callsite, but that is the best we can do.

Reply:
Done